packages/tx/src/types.ts

```diff
@@ -134,6 +138,21 @@ export interface AccessListEIP2930TxData extends TxData {
   type?: BNLike
 }
 
+export interface FeeMarketEIP1559TxData extends AccessListEIP2930TxData {
```

Comment:
Have tested this `never` type, this actually works: 😄 

![image](https://user-images.githubusercontent.com/931137/117289727-3a6fb600-ae6d-11eb-8145-28ab412f4112.png)

Reply:
This had some interesting cascade: I added this in 85c7efc, which caused the type casting in `TransactionFactory.fromTxData()` not to work any more, which I think is a good thing since this was pretty loose on casting everything to an AL tx, which is neither accurate or future-proof. I then reworked this code with type-specific casts and at the end also deprecated the `getTransactionClass()` method.

Might be a case for discussion, but I would be a strong proponent. We are not using this internally with this change any more and this method generally has a somewhat strange API, I would assume that this won't be much used by external users.